flink-java/src/main/java/org/apache/flink/api/java/DataSet.java

```diff
@@ -394,6 +396,21 @@ public long count() throws Exception {
 		return res.<Long> getAccumulatorResult(id);
 	}
 
+	/**
+	 * Convenience method to get the count (number of elements) of a DataSet
+	 * as well as the checksum (sum over element hashes).
+	 *
+	 * @return A Checksum that represents the count and checksum of elements in the data set.
+	 */
+	public Checksum checksum() throws Exception {
+		final String id = new AbstractID().toString();
+
+		flatMap(new Utils.ChecksumHelper<T>(id)).name("checksum()")
+				.output(new DiscardingOutputFormat<NullValue>()).name("checksum() sink");
```

Comment:
This trick to have a `flatMap()` function and then a discarding sink comes from earlier versions of Flink, where Sinks could not use Accumulators.

Since RichSinkFunctions can use accumulators, it would be good to actually have the Checksum helper as a sink.